packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -93,47 +98,91 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
         this.client = client;
     }
 
-    protected getArgs(options?: SearchInWorkspaceOptions): string[] {
-        const args = ['--hidden', '--json'];
-        args.push(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
+    protected getArgs(rootPaths: string[], options?: SearchInWorkspaceOptions): string[] {
+        const args = new Set<string>();
+
+        const appendGlobArgs = (rawPatterns: string[], exclude: boolean) => {
+            rootPaths.forEach(root => {
+                for (const rawPattern of rawPatterns) {
+                    if (rawPattern !== '') {
+                        const globArguments = this.pathToGlobs(rawPattern, root, exclude);
+                        globArguments.forEach(arg => args.add(arg));
+                    }
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        args.add(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
```

Comment:
I noticed the following as well when we execute `rg`, do you know why we need the trailing `--`?

Ex (searching for _about_):

```
root INFO processOptions: command = /home/evinfug/workspaces/theia/node_modules/vscode-ripgrep/bin/rg
root INFO processOptions: args = 
--hidden
--json
--ignore-case
--max-filesize=20M
--glob=!**/**/.git
--glob=!**/**/.svn
--glob=!**/**/.hg
--glob=!**/**/CVS
--glob=!**/**/.DS_Store
--fixed-strings
--
about
/home/evinfug/workspaces/theia
```

Reply:
@alvsan09 thank you for following up! 
Given that description shouldn't `--` be added in all cases and not solely in the `else`?

Reply:
I tried it, and it unfortunately break the behavior as several unit test fail